### PR TITLE
Added support for Android dynamic feature plugins

### LINKED ISSUE #441: Are Android dynamic feature modules supported
Hi, does Kover support Android Dynamic Feature Modules? When I add the plugin to a DFM in my multiproject and try to run `feature:dynamicFeature:koverHtmlReportDebug`, I get an error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'koverHtmlReportDebug' not found in project ':feature:dynamicFeature'.
```

If I try to create a multiproject report in my app module, by using `kover(project(":feature:dynamicFeature"))` and running `:app:koverHtmlReportDebug`, I get below error:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:koverHtmlReportDebug'.
> Could not resolve all task dependencies for configuration ':app:koverExternalArtifactsDebug'.
   > Could not resolve project :feature:dynamicFeature.
     Required by:
         project :app
      > The consumer was configured to find attribute 'kotlinx.kover.android.build-type' with value 'debug'. However we cannot choose between the following variants of project :feature:dynamicFeature:
          - benchmarkApiElements
          - benchmarkReverseMetadataElements
          - benchmarkRuntimeElements
          - debugApiElements
          - debugReverseMetadataElements
          - debugRuntimeElements
          - koverArtifact
          - releaseApiElements
          - releaseReverseMetadataElements
          - releaseRuntimeElements
        All of them match the consumer attributes:
          - Variant 'benchmarkApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'benchmarkReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'benchmarkRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'benchmark' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'benchmark' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'debugApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'debugReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'debugRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'koverArtifact' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides kotlinx.kover.project.path ':feature:dynamicFeature' but the consumer didn't ask for it
                  - Provides kotlinx.kover.variant.name '' but the consumer didn't ask for it
          - Variant 'releaseApiElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'releaseReverseMetadataElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.usage 'android-reverse-meta-data' but the consumer didn't ask for it
          - Variant 'releaseRuntimeElements' capability myProject.feature:dynamicFeature:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.3.1' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.android.build-type (required 'debug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
```

The plugin works as expected for all non-DFM modules in my project.

### PR DESCRIPTION
**Pull request by shanshin**
Resolves #441

### CODE REVIEW

File: kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+plugins {
+    id ("com.android.library")
```
Comment (saihemanth121) [inline kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts:2]: Shouldn't this be a dynamic feature module in this example?

Comment (shanshin) [reply kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts:2]: Yes, thanks!
My bad)

File: kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+
```
Comment (gabrielprado-mb) [inline kover-gradle-plugin/examples/android/dynamic/dyn/src/main/AndroidManifest.xml:4]: To make this module work as a Dynamic Feature, you must declare the <dist> tag and the install type.

Like this one:
```
<dist:module
    dist:instant="false"
    dist:title="@string/mgm_title">
    <dist:delivery>
        <dist:install-time />
    </dist:delivery>
    <dist:fusing dist:include="false" />
</dist:module>
```

File: kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+    implementation("androidx.core:core-ktx:1.8.0")
+    implementation("androidx.appcompat:appcompat:1.5.0")
+    testImplementation("junit:junit:4.13.2")
+}
```
Comment (gabrielprado-mb) [inline kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts:36]: You must add the base (:app) module dependency here. 

```suggestion
    implementation(project(":app"))
}
```

File: kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        viewBinding = true
+    }
+}
```
Comment (gabrielprado-mb) [inline kover-gradle-plugin/examples/android/dynamic/app/build.gradle.kts:36]: You must add the dynamicFeatures tag with your module here or it won't work as Dynamic Feature Module.
```suggestion
     dynamicFeatures = [":dyn"]
}
```

File: kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts (hunk 1 above)
Comment (sandwwraith) [reply kover-gradle-plugin/examples/android/dynamic/dyn/build.gradle.kts:2]: Yes, looks like plugin should be `com.android.dynamic-feature`

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: kover-gradle-plugin/examples/android/dynamic/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+plugins {
+    id("com.android.application") version "7.4.0" apply false
+    id("com.android.library") version "7.4.0" apply false
```
Comment (sandwwraith) [inline kover-gradle-plugin/examples/android/dynamic/build.gradle.kts:3]: Shouldn't `"com.android.dynamic-feature"` also be here with version?

**Review by sandwwraith — APPROVED** [review]

File: kover-gradle-plugin/examples/android/dynamic/build.gradle.kts (hunk 5 above)
Comment (shanshin) [reply kover-gradle-plugin/examples/android/dynamic/build.gradle.kts:3]: I think this version got implicitly together with other plugins, so it did not lead to errors. But for clarity I will add

**Pull request merged**